CIPs/cip-draft_mnemonic_error_correcting_codes.md

```diff
@@ -0,0 +1,233 @@
+* If the result of error correction is a different phrase, then the given phrase contained at least
+  one error, but was recovered successfully. 
+  * Note that error correction will change at most `floor((N - K) / 2)` unknown positions.
+* If error correction fails, then the phrase is either not a CIP-# phrase, or has too many errors to
+  fix.
+
+<!-- TODO(victor) Analyse the probability of this kind of failure. Its likely similar to the issue
+of mistakenly identifying a plain BIP-39 phrase as a CIP-# phrase, which is described below. -->
+Note that if the given phrase contains more than `floor((N - K) / 2)`, then there is a small
+probability that error correction will report success and return a phrase different than the true
+source phrase. Given the chance of mistaken "correction" of a phrase, applications SHOULD inform
+the user before proceeding with a phrase altered by this error correction procedure.
+
+#### Acceptance of plain BIP-39 phrases by CIP-# applications
+
+Applications that implement this proposal SHOULD also accept plain BIP-39 phrases. A phrase SHOULD
+be treated as a plain BIP-39 phrase if error correction fails and the given phrase has a valid
+BIP-39 checksum. Although it is technically possible that a plain BIP-39 phrase may pass error
+correction, it is very unlikely.
+
+## Rationale
+
+### Probability of mistaken identification of a plain BIP-39 phrase as a CIP-# phrase
+
+Described above, error correction and a checksum validation are used to determine if error
+correction is available. It is techically possible, but unlikely, for this to mistake a plain BIP-39
+phrase for a CIP-# phrase. Specifically, given a plain BIP-39 phrase of `N` words, `e` of which are
+invalid (i.e. erasures), the probability that the phrase will pass error correction is equal to the
+probability that:
+
+* A "quorum" subset of symbols (i.e. points) are consistent with some `K-1` degree polynomial.
+* And the re-encoded codeword contains a valid BIP-39 checksum.
+
+<!-- TOOD(victor) Refactor the math in this section. It is _not_ easy to read -->
+
+In this case, the quorum of points required is the number of points available minus the number of
+acceptable unknown errors :`(N - e) - floor((N - K - e) / 2) = ceil((N + K - e) / 2)`. Given a
+particular quorum, it is consistent with some `K-1` degree polynomial if and only if, taking any `K`
+points, the remaining `ceil((N - K - e) / 2)` points are consistent with the polynomial interpolated
+from those `K` points. When  the points are uniformly random and independent of each other the
+probability that this is true is `y = 2 ^ (-11 * ceil((N - K - e) / 2))`. (11 because the points are
+in `GF(2^11)`)
+
+In a given phrase, there are `(N - e) choose ceil((N + K - e) / 2)` quorums. When the words of the
+phrase are chosen independently and at random, The overall chance of decoding a plain BIP-39 phrase
+is `1 - (1 - y) ^ ((N - e) choose ceil((N + K - e) / 2))`, with `y` defined above.
```

Comment:
Nice to add - "meaning, at least one succeeds".

Reply:
Added a clause to that effect